interbotix_common_toolbox/interbotix_gravity_compensation/src/interbotix_gravity_compensation.cpp

```diff
@@ -0,0 +1,524 @@
+  for (size_t i = 0; i < num_joints_arm_; i++) {
+    command_msg.cmd[i] /= current_units_[i];
+  }
+
+  // Publish the JointGroupCommand message if gravity compensation is enabled
+  enable_mutex_.lock();
+  if (gravity_compensation_enabled_) {
+    joint_group_pub_->publish(command_msg);
+  }
+  enable_mutex_.unlock();
+}
+
+void InterbotixGravityCompensation::gravity_compensation_enable_cb(
+  const std::shared_ptr<std_srvs::srv::SetBool::Request> request,
+  std::shared_ptr<std_srvs::srv::SetBool::Response> response)
+{
+  if (request->data) {
+    // Set the operating mode to 'current' if the joints support it
+    set_operating_modes("group", arm_group_name_, "current");
+    set_operating_modes("single", gripper_joint_name_, "current");
+  } else {
+    // Set false the gravity compensation flag
+    enable_mutex_.lock();
+    gravity_compensation_enabled_ = false;
+    enable_mutex_.unlock();
+
+    // Set the operating mode to 'position' for all joints
+    set_operating_modes("group", arm_group_name_, "position");
+    set_operating_modes("single", gripper_joint_name_, "current_based_position");
+  }
+
+  if (request->data) {
+    // Enable the torque for the joints if they support current control mode
+    for (size_t i = 0; i < joint_names_.size(); i++) {
+      if (i < num_joints_arm_) {
+        // Arm joints
+        if (torque_constants_[i] == -1) {
+          // Disable if the joint does not support current control
+          torque_enable("single", joint_names_[i], false);
+        } else {
+          // Enable if the joint supports current control
+          torque_enable("single", joint_names_[i], true);
+        }
+      } else {
+        // Gripper joint
+        torque_enable("single", joint_names_[i], false);
+      }
+    }
+
+    // Set true the gravity compensation flag
+    enable_mutex_.lock();
+    gravity_compensation_enabled_ = true;
+    enable_mutex_.unlock();
+  } else {
+    // Enable the torque for all joints
+    for (size_t i = 0; i < joint_names_.size(); i++) {
+      torque_enable("single", joint_names_[i], true);
+    }
+  }
+
+  // Set the gravity compensation flag and return the response
+  response->success = true;
+  response->message = std::string("Gravity Compensation ");
+  response->message += request->data ? "Enabled" : "Disabled";
+  RCLCPP_INFO(this->get_logger(), response->message.c_str());
+}
+
+bool InterbotixGravityCompensation::load_motor_specs(const std::string & motor_specs)
+{
+  // Load the torque constants and current units from the motor_specs file
+  YAML::Node motor_specs_node;
+  try {
+    motor_specs_node = YAML::LoadFile(motor_specs);
+  } catch (YAML::Exception & e) {
+    RCLCPP_ERROR(
+      this->get_logger(),
+      "Failed to load the motor specs file '%s'! Error: %s",
+      motor_specs.c_str(), e.what());
+    return false;
+  }
+
+  // Load the torque constants, current units, and no load currents
+  float all = motor_specs_node["motor_assist"]["all"].as<float>();
```

Comment:
You can add defaults/fallbacks like this:

```suggestion
  float all = motor_specs_node["motor_assist"]["all"].as<float>(-1.0);
```

I would recommend doing this for all where relevant (e.g. setting dither to `false` by default.